### DISCUSSION TITLE
Right definition file for Daikin Rotex HPSU Compact 508 H/C BIV

### DISCUSSION THREAD

**Discussion by DanieleNeo** [raomin/ESPAltherma#481] [Q&A]
Hi,

I searched around and found this old thread where it says the correct file is ALTHERMA(HPSU6_ULTRA).h
https://github.com/raomin/ESPAltherma/issues/45

I looked at the wiki and found the same thing
https://github.com/raomin/ESPAltherma/wiki/Choosing-the-right-definition-file

The problem is that this file doesn't exist anymore (probably it has been replaced) and I have no idea what is the most correct definition for my machine. I am currently using default.h and it works, but I am missing some values ​​to calculate the COP correctly, can someone help me?

Thanks,
D.

**Comment by philipgo** [comment]
I have a 2011 Rotex HPSU 508 Compact. I use the PROTOCOL_S_ROTEX.h file with some modifications. Essentially it only spits out these values: (my interpretation in german added as comments)
```
{0x53, 0, 200, 1, -1, "Circulation pump"}, //Rotex Zirkulationspumpe
{0x53, 3, 200, 1, -1, "External heater?"}, //Solarunterstützung
{0x53, 5, 200, 1, -1, "Priority to domestic water"},//Warmwasserbereitung aktiv
{0x53, 6, 200, 1, -1, "Burner inhibit from solaris"},// ?
{0x53, 0, 152, 2, -1, "ProtS EV (pls)"}, // ? value is always 256
{0x53, 2, 164, 1, -1, "ProtS Outdoor Fan (Upper)(rps)"}, // ?  value always 0
{0x53, 3, 164, 1, -1, "ProtS Outdoor Fan (Lower)(rps)"}, // ? value always 0
{0x53, 4, 200, 1, -1, "ProtS INV Comp. Frequency(Hz)"}, // ? value always off
{0x53, 8, 200, 1, -1, "ProtS 20R (SV) Output"}, / 
{0x53, 7, 200, 1, -1, "ProtS 20S (4-way) Output"},
{0x53, 10, 200, 1, -1, "ProtS Crankcase Heater"},
0x53, 11, 200, 1, -1, "ProtS Ener-Cut Output"},
{0x54, 0, 153, 2, -1, "Refrig. Temp. liquid side(C)"},
{0x54, 2, 153, 2, -1, "Inlet water temp.(C)"}, // Temperatur Rücklauf
{0x54, 4, 153, 2, -1, "Outlet Water Temp.(C)"}, // Temperatur Vorlauf
{0x54, 6, 153, 2, -1, "D(C)"}, // ?
{0x54, 8, 153, 2, -1, "DHW tank temp.(C)"}, // Temperatur Warmwasser
{0x54, 10, 103, 2, -1, "F(C)"}, // ?
{0x54, 12, 101, 1, -1, "Delta-Tr(deg)"}, // Zieltemperatur Vorlauf Fußbodenheizung
{0x54, 13, 151, 1, -1, "R/C Setpoint(C)"}, // ?
{0x55, 0, 201, 1, -1, "Operation Mode"}, // Heizen oder Kühlen
{0x55, 1, 204, 1, -1, "Error Code"},
{0x55, 2, 204, 1, -1, "Thermo Off Error"},
{0x55, 3, 204, 1, -1, "Warning Code"},
{0x55, 4, 204, 1, -1, "Caution Code"},
{0x54, 6, 103, 2, -1, "ProtS Outdoor heat exchanger temp.(C)"},
{0x54, 8, 109, 2, -1, "ProtS Discharge pipe temp.(C)"},
{0x54, 12, 312, 1, -1, "ProtS Delta-Tr(deg)"},
{0x56, 0, 103, 2, -1, "????"}, // Warmwasser Zieltemperatur
```

Maybe we can identify more values :)

**Reply by DanieleNeo** [reply]
Hi @philipgo,

I tried the definition you said but I think that it's not good for me, some values are surely wrong, operation mode for example is "heating" when my heat pump is off because I'm waiting for the first start (scheduled at 5/12, I bought the heat pump a few years ago and only this year I managed to put it into operation). Mine is 2014 version, so something is different I think. From what I understand Daikin Rotex HPSU Compact is "the same" as Altherma ECH2O (the first versions at least), I used the file def/Altherma(ERGA E EHSH-X P30-50 E_EF series 04-08kW-ECH2O).h and some values ​​come back to me

`Operation Mode
Fan Only
Defrost Operation
OFF
R1T-Outdoor air temp.
3
INV primary current (A)
0,5
INV secondary current (A)
0
INV frequency (rps)
0
I/U operation mode
Stop
Thermostat ON/OFF
OFF
Freeze Protection
OFF
DHW setpoint
25
LW setpoint (main)
15
Water flow switch
OFF
2way valve(On:Heat Off:Cool)
ON
3way valve(On:DHW Off:Space)
OFF
BUH Step1
OFF
BUH Step2
OFF
Leaving water temp. before BUH (R1T)
12,9
Leaving water temp. after BUH (R2T)
12,7
Refrig. Temp. liquid side (R3T)
12,6
Inlet water temp.(R4T)
12,7
DHW tank temp. (R5T)
12,1
Indoor ambient temp. (R1T)
0
Ext. indoor ambient sensor (R6T)
8,8
WifiRSSI
-87dBm
FreeMem
224944`

I'll do some tuning in the next few days...

**Reply by philipgo** [reply]
> operation mode for example is "heating" when my heat pump is off because I'm waiting for the first start

This can be correct, it is only the chosen mode, no matter whether the pump is actually heating or not.

Nevertheless, your pump most likely does not have Protocol S anymore.

**Reply by DanieleNeo** [reply]
I don't know if it can be correct, the "fan only" in my opinion it is more suitable for a device in idle. 

Anyway @philipgo, wandering around I came across this German project that uses the CAN bus to extract information and modify settings, all based on ESPHome

https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN-Seriell/tree/main

I don't have the hardware at the moment so I'm playing around with the software, and it *seems* (I can't be sure yet because I have a faulty sd card on my raspberry that I need to change and which therefore restarts the device during build phase) that it build directly from ESPHome, which is great!

One thing that I will definitely have to do is modify the template for an Italian or at least international localization, you have an advantage in this :)

It seems that on our Daikin/Rotex it pulls out all the information, allows you to make the settings, and if I understood correctly it should, always via CAN bus, activate/deactivate the heating/cooling, a slightly more digital approach than the mixed digital/analog of ESPAltherma + relay.

At the moment I have an ESPAltherma setup that I will keep, at the same time I will test this new solution, then I will decide. I thought this sharing could be useful to you.
